src/spikeinterface/sorters/launcher.py

```diff
@@ -146,12 +155,18 @@ def run_sorter_jobs(job_list, engine="loop", engine_kwargs={}, return_output=Fal
 
     elif engine == "slurm":
         # generate python script for slurm
-        tmp_script_folder = engine_kwargs["tmp_script_folder"]
+        tmp_script_folder = engine_kwargs.pop("tmp_script_folder")
         if tmp_script_folder is None:
             tmp_script_folder = tempfile.mkdtemp(prefix="spikeinterface_slurm_")
         tmp_script_folder = Path(tmp_script_folder)
-        cpus_per_task = engine_kwargs["cpus_per_task"]
-        mem = engine_kwargs["mem"]
+        sbatch_executable = engine_kwargs.pop("sbatch_executable_path")
+
+        # for backward compatibility with previous version
+        if "cpus_per_task" in engine_kwargs:
+            warnings.warn("cpus_per_task is deprecated, use cpus-per-task instead", DeprecationWarning)
+            cpus_per_task = engine_kwargs.pop("cpus_per_task")
+            if "cpus-per-task" not in engine_kwargs:
```

Comment:
I wonder if better (though a bit more verbose) to raise an error here, if the user has accidentally passed two versions of the same kwarg it would be better to let them know I think.